AmplifyPlugins/DataStore/Tests/DataStoreHostApp/AWSDataStorePluginAuthCognitoTests/AWSDataStoreAuthBaseTest.swift

```diff
@@ -424,40 +382,31 @@ extension AWSDataStoreAuthBaseTest {
                 }
 
                 if mutationEvent.mutationType == GraphQLMutationType.create.rawValue {
-                    expectations.mutationSaveProcessed.fulfill()
+                    Task { await expectations.mutationSaveProcessed.fulfill() }
                     return
                 }
 
                 if mutationEvent.mutationType == GraphQLMutationType.delete.rawValue {
-                    expectations.mutationDeleteProcessed.fulfill()
+                    Task { await expectations.mutationDeleteProcessed.fulfill() }
                     return
                 }
             }
             .store(in: &requests)
-
-        Amplify.DataStore.save(model).sink {
-            if case let .failure(error) = $0 {
-                onFailure(error)
+        Task {
+            do {
+                let posts = try await Amplify.DataStore.save(model)
+                XCTAssertNotNil(posts)
+                let deletedposts: () = try await Amplify.DataStore.delete(posts)
+                XCTAssertNotNil(deletedposts)
+            } catch crudError.notDeleted(let error) {
+                XCTFail("Unable to delete posts from Datastore with errors: \(String(describing: error))")
+            } catch crudError.notSaved(let error) {
+                XCTFail("Unable to save posts from Datastore with errors: \(String(describing: error))")
             }
```

Comment:
DataStore.save and DataStore.delete both throw DataStore error, so you can catch it like this:
```
} catch let error as DataStoreError {
```